Implement design for Set/Reset Password Page

aaleksaaleksic opened the pull request:
Closes: #9

markocic commented on src/app/auth/reset-password/page.tsx:
```diff
@@ -1,3 +1,98 @@
+  .object({
+    password: z
+      .string()
+      .min(8, 'Password must be at least 8 characters')
```
We are not going to use our custom messages since that can lead to inconsistencies across the app and we have to maintain them. If you remove this message zod will provide us with the default message that works just fine. There are exceptions to this tho (e.g. `.regex()`).

markocic commented on src/app/auth/reset-password/page.tsx:
```diff
@@ -1,3 +1,98 @@
+    confirmPassword: z.string(),
+  })
+  .refine((data) => data.password === data.confirmPassword, {
+    message: 'Passwords do not match',
```
This is one of the exceptions I mentioned below. Here the message is fine since zod can't interpret what is the purpose of the refine.

markocic commented on src/app/auth/reset-password/page.tsx:
```diff
@@ -1,3 +1,98 @@
+        </CardHeader>
+
+        <CardContent className="px-3 pb-3">
+          <form onSubmit={handleSubmit(onSubmit)} className="space-y-4">
```
We should use `Form` component from shadcn. Please refer to the [documentation](https://ui.shadcn.com/docs/components/form) on its usage and edit the appropriate code.

markocic commented on src/app/auth/reset-password/page.tsx:
```diff
@@ -1,3 +1,98 @@
+  return (
+    <div className="flex min-h-screen items-center justify-center bg-gray-100">
+      <Card className="w-full max-w-[348px] max-h-[348px] bg-white rounded-lg border border-zinc-200 p-3">
+        <CardHeader className="w-full h-[100px] p-3 text-left">
```
We shouldn't put height constraints on a component like this. Instead use `padding` to get the desired look. We don't have to follow design to the pixel.

markocic commented on src/app/auth/reset-password/page.tsx:
```diff
@@ -1,3 +1,98 @@
+
+  return (
+    <div className="flex min-h-screen items-center justify-center bg-gray-100">
+      <Card className="w-full max-w-[348px] max-h-[348px] bg-white rounded-lg border border-zinc-200 p-3">
```
`max-w-[348px]` is fine but `max-h-[348px]` could cause future problems. Please remove the height constraint.

markocic commented on src/app/auth/reset-password/page.tsx:
```diff
@@ -1,3 +1,98 @@
+    <div className="flex min-h-screen items-center justify-center bg-gray-100">
+      <Card className="w-full max-w-[348px] max-h-[348px] bg-white rounded-lg border border-zinc-200 p-3">
+        <CardHeader className="w-full h-[100px] p-3 text-left">
+          <h2 className="text-2xl font-semibold text-[#09090B] font-inter">
```
We shouldn't use colors like this. Check figma right sidebar for the exact color name and then refer to it by class. We are going to support light/dark theme so using colors like this will make that harder.

This also refers to all the times you used colors like this.

markocic commented on src/app/auth/reset-password/page.tsx:
```diff
@@ -1,3 +1,98 @@
+            <div className="flex justify-end mt-2">
+              <Button
+                type="submit"
+                className="bg-[#18181B] text-white rounded-md py-1 px-2 w-[80px] h-[40px] text-sm font-medium"
```
Please remove `width` and `height` constrains.

markocic commented on src/app/layout.tsx:
```diff
@@ -2,12 +2,19 @@ import type { Metadata } from 'next';
 import { Geist, Geist_Mono } from 'next/font/google';
 import './globals.css';
 import { ThemeProvider } from '@/components/theme-provider';
+import { Inter } from 'next/font/google';
```
This was an oversight on my part for not setting the font the same on figma and here. We are going to keep `Geist` for now. You can remove `Inter`.

markocic commented on src/app/page.tsx:
```diff
@@ -1,3 +1,5 @@
+
 export default function Home() {
-  return <h1>Landing page</h1>;
+  return <SetPasswordPage />;
```
Rollback this file.

markocic commented on src/app/auth/reset-password/page.tsx:
```diff
@@ -1,3 +1,98 @@
+  };
+
+  return (
+    <div className="flex min-h-screen items-center justify-center bg-gray-100">
```
Remove `bg-gray-100` to match design.

markocic reviewed (changes requested):
Great work! Minor changes before this can be merged.

aaleksaaleksic commented:
I've addressed all the feedback and implemented the requested changes. I've also ensured that everything is properly rebased with the latest changes from main. Let me know if there's anything else you'd like me to adjust.

Thanks for the detailed feedback! Looking forward to your review.

markocic commented on src/app/auth/password/[type]/[token]/page.tsx:
```diff
@@ -0,0 +1,129 @@
+export default function PasswordPage() {
+  const params = useParams<PasswordParams>();
+
+  if (!params?.type || !params?.token) {
```
This kind of check is not needed. If you see type of params it is `PasswordParams`.

markocic reviewed (changes requested):
Thank you for the commits! Just one minor change and this is ready to merge.

aaleksaaleksic commented:
Just made the last change you requested! Hope everything is fine now! 😁
